packages/ui/lib/styled/Pagination.js

```diff
@@ -104,14 +123,17 @@ export default function Pagination({
           <PageItem
             key={index}
             active={page === index + 1}
-            onClick={() => setPage(index + 1)}
+            onClick={() => handleChange(index + 1)}
           >
-            {index + 1}
+            {itemRender(index + 1, "page", <a>{index + 1}</a>)}
           </PageItem>
         )
       )}
-      <Nav disabled={page === totalPages} onClick={() => setPage(page + 1)}>
-        <CaretRight />
+      <Nav
+        disabled={page === totalPages}
```

Comment:
`disabled={page >= totalPages}` would be better since upper layer caller might passed in wrong `page`